src/metabase/models/setting.clj

```diff
@@ -780,7 +787,12 @@
   ###### `:database-local`
 
   The ability of this Setting to be /Database-local/. Valid values are `:only`, `:allowed`, and `:never`. Default:
-  `:never`. See docstring for [[metabase.models.setting]] for more information."
+  `:never`. See docstring for [[metabase.models.setting]] for more information.
+
+  ###### `:deprecated`
+
+  If this setting is deprecated, this should contain a string of the Metabase version in which the setting was
+  deprecated. A deprecation notice will be logged whenever the setting is written. (Default: `nil`)."
```

Comment:
@camsaul want to call your attention to this; it seemed useful here (similar to `:deprecated` metadata on functions) but let me know what you think

Reply:
I like it. You could update `metadata-for-setting-fn` to make the metadata for the setting function include the deprecation. Static things won't benefit from this but nrepl will make the call sites visibly different.